### PR TITLE
SOLR-15567: Document Schema Designer screen in ref guide.

### PR DESCRIPTION
**Pull request by thelabdude**
https://issues.apache.org/jira/browse/SOLR-15567

Documentation for the Schema Designer

### CODE REVIEW

**Comment by ctargett** [comment]
With the changes I made in the recent big re-org, one thing I did was break up all the Admin UI pages and spread them around so they're adjacent to the docs for whatever the UI provides access to - the Cloud screens docs is with the other SolrCloud content, the Logging screen docs got merged with the docs on how to configure logging, the Document Upload screen is with the pages about indexing, etc.

Basically, I wanted to stop treating the Admin UI as a special, separate thing, and instead document overall features/functions more holistically and in a less disjointed way - here's the way this thing works, here's how you interact with Solr to configure this thing, including a UI for it (when there is one). Depending on how complex the "thing" is, it might be all on a single page or on multiple pages.

So following this organizational idea, I think the Schema Designer should move from being a child of `solr-admin-ui.adoc` (where it feels a little orphaned organizationally, being the only child page there) to being a child of either `solr-schema.adoc` or `fields-schema-design.adoc` - that's where all the content about Solr's schema generally and the specifics of the schema elements live now.

I understand why you may have thought to put it under `solr-admin-ui.adoc`, since that page lives under Getting Started, and you stated you wanted the Schema Designer to be a getting started tool. Moving it may seem like it prevents that, but I think we could highlight within the various pages under Getting Started (like in `documents-fields-schema-design.adoc`, for one, but also others) that the tool exists. Then when people do use the docs, they have access through the left nav to the related ideas and documentation that they need. Eventually someday we/someone could write a tutorial which would walk a new user through it using sample data.

What do you think?

**Comment by thelabdude** [comment]
Sounds great, thanks for the guidance here. Will move it under `fields-schema-design.adoc`.

**Comment by thelabdude** [comment]
Ended up putting it under `solr-schema` but I kept the link on the `admin-ui` page as well.

File: solr/solr-ref-guide/src/schema-designer.adoc [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+Previously uploaded sample documents are indexed in the temporary collection even though they do not display in the text area.
+
+[TIP]
+====
+Click on the *Edit Documents* button on the *Query Results* panel to load a JSON representation of indexed documents into the text area.
+====
+
+=== Iteratively Post Sample Documents
+
+If you have sample documents spread across multiple files, you can POST them to the Schema Designer API and then load your schema in the Designer UI to design your schema.
```
Comment (ctargett) [inline solr/solr-ref-guide/src/schema-designer.adoc:83]: This API is mentioned a couple of times, most prominently here, but it doesn't seem to be documented in full anywhere?

File: solr/solr-ref-guide/src/schema-designer.adoc [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+done
+----
+
+After sending the sample documents to the Schema Designer `/analyze` endpoint, you can open the schema in the UI in your browser.
+
+== Schema Editor
+
+After analyzing your sample documents, the Schema Designer loads the schema in the *Schema Editor* in the middle panel.
+The editor renders the schema as a tree component composed of Fields, Dynamic Fields, Field Types, and Files.
+For more information about schema objects, see <<fields-and-schema-design.adoc#,Fields and Schema Design>>
```
Comment (ctargett) [inline solr/solr-ref-guide/src/schema-designer.adoc:119]: Need a period to end the sentence.

File: solr/solr-ref-guide/src/schema-designer.adoc [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+* Field guessing (aka "schemaless mode") allows Solr to detect the "best" field type for unknown fields encountered during indexing. Field guessing also performs some field transformations, such as removing spaces from field names. If you use the schema designer to create your schema based on sample documents, you may not need to enable this feature. However, with this feature disabled, you need to make sure the incoming data matches the schema exactly or indexing errors may occur. For more information about schemaless mode, see: <<schemaless-mode.adoc#,Schemaless Mode>>
+* Enabling this feature adds the _root_ and _nest_path_ fields to your schema. For more information about indexing nested child documents, see: <<indexing-nested-documents.adoc#,Indexing Nested Documents>>
```
Comment (ctargett) [inline solr/solr-ref-guide/src/schema-designer.adoc:136]: The `_root_` and `_nest_path_` fields here are getting rendered in HTML in italics. I think if you add backticks to make those monospace you won't need to do any escaping of the underscores.

**Review by ctargett — CHANGES_REQUESTED** [review]
It looks pretty good overall, just a couple things to fix.

File: solr/solr-ref-guide/src/schema-designer.adoc (hunk 1 above)
Comment (thelabdude) [reply solr/solr-ref-guide/src/schema-designer.adoc:83]: right ... in this case, the Schema Designer API is mainly intended to support the UI and not really intended to be used directly. I'll make a note of that here ...

**Review by ctargett — APPROVED** [review]
+1, looks good!

**Pull request merged**